jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -648,8 +648,26 @@ private boolean parseLine(ByteBuffer buffer)
                         _length=_string.length();
                         _methodString=takeString();
                         HttpMethod method=HttpMethod.CACHE.get(_methodString);
-                        if (method!=null)
-                            _methodString=legacyString(_methodString,method.asString());
+                        if (method!=null && method.asString()!=_methodString)
+                        {
+                            // TODO See #1966 Case Sensitive Method
+                            if (method.asString()!=_methodString)
```

Comment:
I don't think this is right. 

First of all, it's the same check done in the outer if, so this line is always `true`. 

Then, it's comparing `String`s with `!=`; `methodString` is always generated from what is being parsed via `takeString()`, so it is always `true` that `method.asString() != _methodString`, no ?

Reply:
oops - I moved the condition to the outer if, so I could get rid on the inner if... but forgot to remove it. Fixed now.

Reply:
Still not convinced by `String` comparisons.